src/libcore/str/mod.rs

```diff
@@ -368,6 +376,38 @@ pub fn next_code_point(bytes: &mut slice::Iter<u8>) -> Option<u32> {
     Some(ch)
 }
 
+/// Reads the last code point out of a byte iterator (assuming a
+/// UTF-8-like encoding).
+#[unstable(feature = "core")]
+#[inline]
+pub fn next_code_point_reverse(bytes: &mut slice::Iter<u8>) -> Option<u32> {
```

Comment:
I mostly marked it public because its exact mirror `next_code_point` is marked public.

Seeing how both do exactly the same thing from different directions, it felt inconsistent to make one of them private.